dpytools/http/upload.py

```diff
@@ -0,0 +1,131 @@
+import datetime
+import os
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Tuple, Union
+
+from dpytools.http.base import BaseHttpClient
+
+
+class UploadClient(BaseHttpClient):
+    def __init__(self, upload_url: str):
+        # Inherit backoff_max value from BaseHTTPClient.__init__
+        super().__init__()
+        self.upload_url = upload_url
+        # For security purposes, `self.access_token` should be set using a FLORENCE_TOKEN environment variable
+        self.access_token = os.environ.get("FLORENCE_TOKEN")
+        self.headers = {"X-Florence-Token": self.access_token}
+
+    # TODO The `upload` method is currently configured to match the old `upload` endpoint specification. Once the `upload-new` endpoint has been released, the `params` argument passed to `self.post` will need to be adjusted to the new specification (see https://github.com/ONSdigital/dp-upload-service/blob/b91b53a13b28c6ba1095abee71680c591862c68c/swagger.yaml).
+    def upload(
+        self,
+        csv_path: Union[Path, str],
+        s3_path: str,
```

Comment:
I don't believe that you can choose which s3 bucket to upload to through the upload api. It will be a set bucket and (annoyingly) the path will differ depending on the environment. So passing s3_path is going to cause issues further down the road because its included in the return variable s3_url (which would be incorrect)

Reply:
Does this mean we would always be able to work out the S3 bucket ID based on the environment? If so, I can just remove this and only return the S3 object key.

Reply:
Yeh thats right.
Not sure what the full upload process is going to look like but if it is similar to CMD then we will need to know the full bucket file path but it doesn't need to come from here

Reply:
we'll set the bucket id as an env var and the script calling this would get it and pass it in.

so switch `s3_path` to `bucket` or similar, and concatenate something like `f"s3://{bucket}/{key}"` in your code. Just need the name of the bucket then.

one quirk of s3 buckets btw - names need to be globally unique, that's why it changes and why they have really long and clunky names with "dp" and env name type info in them 😄